### PR TITLE
[GAC-45] Initial Version of FlightSQL Readme

### PR DESCRIPTION
**Pull request by affonsov**
### Rationale for this change
Non existent FlightSQL readme file

### What changes are included in this PR?
The initial version of FlightSQL readme file

### Are these changes tested?
N/A

### Are there any user-facing changes?
N/A

### CODE REVIEW

File: cpp/src/flightsql_odbc/amazon-timestream-odbc-driver/flightsql-readme.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# FligthSQL ODBC Driver
+
+## Overview 
+
+ODBC driver for Apache Arrow
```
Comment (alinaliBQ) [inline cpp/src/flightsql_odbc/amazon-timestream-odbc-driver/flightsql-readme.md:5]:
```suggestion
ODBC driver for Apache Arrow Flight SQL.
```

File: cpp/src/flightsql_odbc/amazon-timestream-odbc-driver/flightsql-readme.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## Driver Building (Windows)
+
+Open Visual Studio Developer PowerShell or Powershell and run `./build_win32.bat` or `./build_win64.bat` to build the 32-bit and 64-bit driver correspondingly. The .bat file will build and install the dependencies and the driver.
+
+### Build Troubleshoot Guide
+
+| Error | Root Cause | Fix|
+|-------|------------|----|
+| CUSTOMBUILD : CMake error : The source directory "C:/<repo_folder>/build/flight_sql/ApacheArrow-prefix/src/ ApacheArrow" does not appear to contain CMakeLists.txt. [C:\Dev\forked-flightsql-odbc\build\flight_sql\ApacheArrow.vcxp roj] | current CMAKE version 3.28 has issues in build the project | downgrade to the 3.25 version |
```
Comment (alinaliBQ) [inline cpp/src/flightsql_odbc/amazon-timestream-odbc-driver/flightsql-readme.md:27]:
```suggestion
| CUSTOMBUILD : CMake error : The source directory "C:/<repo_folder>/build/flight_sql/ApacheArrow-prefix/src/ApacheArrow" does not appear to contain CMakeLists.txt. [C:\Dev\forked-flightsql-odbc\build\flight_sql\ApacheArrow.vcxproj] | current CMAKE version 3.28 has issues in build the project | downgrade to the 3.25 version |
```
nit - spaces

**Review by alinaliBQ — COMMENTED** [review]
Thanks, I have some minor comments

**Comment by alinaliBQ** [comment]
PR LGTM. Since we have a rebase-and-merge strategy here, please squash the two commits since they are for the same change

File: cpp/src/flightsql_odbc/amazon-timestream-odbc-driver/flightsql-readme.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# FligthSQL ODBC Driver
```
Comment (alinaliBQ) [inline cpp/src/flightsql_odbc/amazon-timestream-odbc-driver/flightsql-readme.md:1]: FYI - after your PR is merged, I can move this read me file as part of my PR #9. Or if #9 is merged first, I might have to ask you to please move this file to `cpp/src/flightsql_odbc/flightsql-odbc`

Comment (alinaliBQ) [reply cpp/src/flightsql_odbc/amazon-timestream-odbc-driver/flightsql-readme.md:1]: #9  is merged, please rebase your branch onto `GAC-odbc-driver` and move the readme to `cpp/src/flightsql_odbc/flightsql-odbc`. 
Tip: On GitHub Desktop, you can drag 2 commits together to form a squash:
https://docs.github.com/en/desktop/managing-commits/squashing-commits-in-github-desktop

image

Comment (affonsov) [reply cpp/src/flightsql_odbc/amazon-timestream-odbc-driver/flightsql-readme.md]: Done

**Pull request closed without merging**